crates/kitsune-type/src/mastodon/notification.rs

```diff
@@ -0,0 +1,28 @@
+use iso8601_timestamp::Timestamp;
+use serde::{Deserialize, Serialize};
+use speedy_uuid::Uuid;
+use utoipa::ToSchema;
+
+use super::{Account, Status};
+
+#[derive(Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum NotificationType {
+    Mention,
+    Status,
+    Reblog,
+    Follow,
+    FollowRequest,
+    Favourite,
+    Update,
+}
+
+#[derive(Clone, Deserialize, Serialize, ToSchema)]
+pub struct Notification {
+    pub id: Uuid,
+    #[serde(rename = "type")]
+    pub _type: NotificationType,
```

Comment:
```suggestion
    pub r#type: NotificationType,
```

And you can remove the serde rename annotation if you apply this

Reply:
oh nice, it's a cool trick, thanks!